identity/auth-starter/src/main/java/io/camunda/identity/authentication/handler/CamundaAuthenticationFailureHandler.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.identity.authentication.handler;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.security.web.access.AccessDeniedHandler;
+import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+import org.springframework.stereotype.Component;
+
+@Component
+public final class CamundaAuthenticationFailureHandler
```

Comment:
There should be no need to name this class `Camunda`, right? Since it is doing authentication failure and authorization failure handling (as in [here](https://github.com/camunda/zeebe/pull/18428/files#diff-5e1ec7eb80d340f958c492a1cee5e647b22898f9b18c5d1f540c7edcb8b7a7aaR59)), may be a name that would fit both is `AuthFailureHandler`?

Reply:
I removed the `camunda` here